File: solution/121.best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int max = 0;
    int temp = 0;
    int i = 0;
    int j;
    for(j = 1;j < pricesSize;j ++)
    {
        if(prices[j] > prices[i])
            temp = prices[j] - prices[i];
        else
            i = j;
        if(temp > max)
            max = temp;
    }
    return max;
}

File: solution/121.best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let max = 0,
    temp = 0;
  for (let i = 0, j = 1; j < prices.length; j++) {
    if (prices[j] > prices[i]) {
      temp = prices[j] - prices[i];
    } else {
      i = j;
    }
    if (temp > max) {
      max = temp;
    }
  }
  return max;
};